content/behind_the_scenes/index.md

```diff
@@ -0,0 +1,124 @@
+
+[Stunnel][stunnel] is used to ensure that the data you send to us is encrypted and tamper resistant. Furthermore, we send the feeder's plane.watch API key in the TLS request's [Server Name Indication (SNI)][sni] field, allowing us to determine which data is associated to which feeder.
+
+### bordercontrol ###
+
+**Bordercontrol** is our custom-written feed receiver. Its job is to handle the many incoming connections from feeders running our [pw-feeder][pw-feeder] client software.
+
+When a [pw-feeder][pw-feeder] client connects:
+
+1. The feeder's API key is read from the stunnel's [SNI][sni] information.
+2. Bordercontrol queries [ATC](#ads-b-traffic-control-atc) to determine if the API key matches that of a valid feeder.
+3. If the API key is valid, a unique "feed-in" container is spun up for that client. If the API key is invalid, the connection is dropped.
+4. The incoming [BEAST][beast protocol] connection is proxied to the feed-in container.
+5. The incoming [MLAT][mlat] connection is proxied to an instance of [mlat-server][mlat-server] running on the regional multiplexer.
+
+### pw_ingest ###
+
+[**pw_ingest**][pw-pipeline] runs on each feed-in container. It decodes [BEAST][beast protocol] protocol frames, and publishes the data as messages to [NATS][nats], for processing through the [plane.watch pipeline][pw-pipeline]. Messages are tagged with the feeder's API key, so the source of the data can be tracked throughout the processing pipeline.
+
+### pw-enrichment ###
+
+**pw-enrichment**, also known as EC (enrichment centre), consumes decoded ADS-B frames from the [NATS][nats] message queue and adds additional information relating to the aircraft, route and other metadata. The EC maintains an internal memory-cache so as to not overwhelm the ATC database, given the message rate can be in the thousands of messages a second.
+
+### pw_router ###
+
+[**pw_router**][pw-pipeline] takes enriched data and reduces it down to significant events, thus cutting the message rate significantly. The router also publishes two feeds to the NATS bus - a high resolution and a low resolution - these are used in the UI to not overwhelm the browser with messages.
+
+Router also connects to an instance of [ClickHouse], a highly scalable columnar database designed for storing timeseries data - perfect for ADSB. Both high and low resolution feeds are stored to Clickhouse for later analysis and for drawing the historical paths in the UI.
+
+### pw_ws_broker ###
+
+[**pw_ws_broker**][pw-pipeline] is a horizontally scalable service that provides a [WebSocket][websocket] interface to connect the public web interface with the backend pipeline. When a user loads the public website, a websocket is dialed up to this service to stream ADSB updates from the pipeline directly to the user.
+
+The broker also provides streaming APIs for querying ATC for more detailed route and airframe information as well as querying ClickHouse for historical track information.
```

Comment:
The broker also provides streaming APIs for querying ATC for more detailed route and airframe information, as well as querying ClickHouse for historical track information.